From 3577db110e03985d4486cae6caf3ceb5696c45e9 Mon Sep 17 00:00:00 2001
From: Jon TURNEY <jon.turney@dronecode.org.uk>
Date: Thu, 20 Dec 2012 22:56:28 +0000
Subject: Fix compilation when NDEBUG is defined

XORG_DEFAULT_OPTIONS enables -Werror=return-type

xcb_image_get_pixel() and xcb_image_create_native() assume assert(0) is
noreturn, and don't return a value in that case.  But assert(0) isn't noreturn
if NDEBUG is defined.

Fix this by returning a suitable value.

(Also, on the MinGW target, assert(0) is non-standard and may return even in the
!NDEBUG case, as the C runtime may ask the user if they want to abort or not...)

/jhbuild/checkout/xcb/xcb-util-image/image/xcb_image.c: In function 'xcb_image_get_pixel':
/jhbuild/checkout/xcb/xcb-util-image/image/xcb_image.c:754:1: error: control reaches end of non-void function [-Wreturn-type]
/jhbuild/checkout/xcb/xcb-util-image/image/xcb_image.c: In function 'xcb_image_create_native':
/jhbuild/checkout/xcb/xcb-util-image/image/xcb_image.c:190:1: error: control reaches end of non-void function [-Wreturn-type]

Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
Signed-off-by: Peter Harris <pharris@opentext.com>
---
diff --git a/image/xcb_image.c b/image/xcb_image.c
index e426cbd..101aa4f 100644
--- a/image/xcb_image.c
+++ b/image/xcb_image.c
@@ -187,6 +187,7 @@ xcb_image_create_native (xcb_connection_t *  c,
       assert(0);
   }
   assert(0);
+  return NULL;
 }
 
 
@@ -751,6 +752,7 @@ xcb_image_get_pixel (xcb_image_t *image,
   default:
       assert(0);
   }
+  return 0;
 }
 
 
--
cgit v0.9.0.2-2-gbebe
